Implement F.pad in ATen

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #73434
* #73433
* #73432
* __->__ #73431

This moves the C++ torch pad function into ATen proper. Once the
forward-compatibility period is over, the python interface can use
this directly.

Differential Revision: [D34571405](https://our.internmc.facebook.com/intern/diff/D34571405)

peterbell10 committed:
Implement F.pad in ATen

This moves the C++ torch pad function into ATen proper. Once the
forward-compatibility period is over, the python interface can use
this directly.

[ghstack-poisoned]

peterbell10 committed:
Update on "Implement F.pad in ATen"


This moves the C++ torch pad function into ATen proper. Once the
forward-compatibility period is over, the python interface can use
this directly.

[ghstack-poisoned]

peterbell10 committed:
Update on "Implement F.pad in ATen"


This moves the C++ torch pad function into ATen proper. Once the
forward-compatibility period is over, the python interface can use
this directly.

[ghstack-poisoned]

peterbell10 committed:
Add comments that python code will be removed on "Implement F.pad in ATen"


This moves the C++ torch pad function into ATen proper. Once the
forward-compatibility period is over, the python interface can use
this directly.

Differential Revision: [D34571405](https://our.internmc.facebook.com/intern/diff/D34571405)

[ghstack-poisoned]

peterbell10 committed:
Update on "Implement F.pad in ATen"


This moves the C++ torch pad function into ATen proper. Once the
forward-compatibility period is over, the python interface can use
this directly.

Differential Revision: [D34571405](https://our.internmc.facebook.com/intern/diff/D34571405)

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "Implement F.pad in ATen"


This moves the C++ torch pad function into ATen proper. Once the
forward-compatibility period is over, the python interface can use
this directly.

Differential Revision: [D34571405](https://our.internmc.facebook.com/intern/diff/D34571405)

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "Implement F.pad in ATen"


This moves the C++ torch pad function into ATen proper. Once the
forward-compatibility period is over, the python interface can use
this directly.

Differential Revision: [D34571405](https://our.internmc.facebook.com/intern/diff/D34571405)

[ghstack-poisoned]

peterbell10 commented:
@ezyang PTAL

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D34571405).

ezyang commented:
needs some warning fixing

```
Summary: 
stderr: caffe2/aten/src/ATen/native/PadNd.cpp:160:26: error: comparison of integers of different signs: 'size_t' (aka 'unsigned long') and 'long' [-Werror,-Wsign-compare]
  TORCH_CHECK(pad.size() <= input_dim * 2, "Padding length too large");
              ~~~~~~~~~~ ^  ~~~~~~~~~~~~~
stderr: caffe2/aten/src/ATen/native/PadNd.cpp:160:26: error: comparison of integers of different signs: 'size_t' (aka 'unsigned long') and 'long' [-Werror,-Wsign-compare]
  TORCH_CHECK(pad.size() <= input_dim * 2, "Padding length too large");
```

peterbell10 committed:
Update on "Implement F.pad in ATen"


This moves the C++ torch pad function into ATen proper. Once the
forward-compatibility period is over, the python interface can use
this directly.

Differential Revision: [D34571405](https://our.internmc.facebook.com/intern/diff/D34571405)

[ghstack-poisoned]

peterbell10 committed:
Rebase on "Implement F.pad in ATen"


This moves the C++ torch pad function into ATen proper. Once the
forward-compatibility period is over, the python interface can use
this directly.

Differential Revision: [D34571405](https://our.internmc.facebook.com/intern/diff/D34571405)

[ghstack-poisoned]
